### PR TITLE
Removed blue formatter from docs an .toml files

### LINKED ISSUE #186: Remove support for blue
As blue is really out of date, we should remove that.
it seems that all the black forks are not really updated anymore ... 
instead, we should offer ruff for linter and for formatter 
that should be a better option for code formatter than black, it also allows single quote option!

### PR DESCRIPTION
**Pull request by Vishalk91-4**
## Pull Request description

Removed Blue formater from docs, auto-commint, pre-commit and various .toml files

Fixes #186 

## How to test these changes

* Run in cli

## Pull Request checklists

This PR is a:
- [x] bug-fix
- [ ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [x] this PR requires a project documentation update.

Author's checklist:
- [x] I have reviewed the changes and it contains no misspelling.
- [x] The code is well commented, especially in the parts that contain more complexity.
- [x] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved
```

### CODE REVIEW

**Comment by Vishalk91-4** [comment]
@Saransh-cpp , could you please review this PR

File: src/scicookie/hooks/post_gen_project.py [hunk 1]
```diff
@@ -259,9 +258,9 @@ def http2ssh(url):
 
 
 def validation():
-    if USE_BLUE and USE_BLACK:
+    if USE_BLACK:
         raise Exception(
-            "The libs Blue and Black were selected, but you need to choose "
+            "The libs Black were selected, but you need to choose "
             "just one of them."
         )
```
Comment (Saransh-cpp) [inline src/scicookie/hooks/post_gen_project.py:265]: This whole `if` is now not required. It was written to make sure that the user does not choose black and blue together.

File: src/scicookie/profiles/base.yaml [hunk 2]
```diff
@@ -61,12 +61,12 @@ project_license:
...
-    - ISC license
-    - Apache Software License 2.0
-    - GNU General Public License v3
-    - Other
+  - BSD 3 Clause
+  - MIT
+  - ISC license
+  - Apache Software License 2.0
+  - GNU General Public License v3
+  - Other
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:69]: This should not have been a part of this PR.

File: src/scicookie/profiles/base.yaml [hunk 3]
```diff
@@ -76,8 +76,8 @@ project_layout:
   default: src
   # first choice is the default for the UI
   choices:
-    - src
-    - flat
+  - src
+  - flat
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:80]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 4]
```diff
@@ -87,15 +87,15 @@ build_system:
...
-    - pybind11
+  - poetry
+  - flit
+  - mesonpy
+  - setuptools
+  - pdm
+  - hatch
+  - maturin
+  - scikit-build-core
+  - pybind11
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:98]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 5]
```diff
@@ -105,9 +105,9 @@ command_line_interface:
   default: None
   # first choice is the default for the UI
   choices:
-    - None
-    - Click
-    - Argparse
+  - None
+  - Click
+  - Argparse
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:110]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 6]
```diff
@@ -117,9 +117,9 @@ documentation_engine:
   default: mkdocs
   # first choice is the default for the UI
   choices:
-    - mkdocs
-    - sphinx
-    - jupyter-book
+  - mkdocs
+  - sphinx
+  - jupyter-book
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:122]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 7]
```diff
@@ -134,24 +134,23 @@ use_tools:
...
+  - isort
+  - make
+  - makim
+  - mccabe
+  - mypy
+  - pre-commit
+  - pydocstyle
+  - pytest
+  - hypothesis
+  - shellcheck
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:152]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 8]
```diff
@@ -161,9 +160,9 @@ use_containers:
   default: None
   # first choice is the default for the UI
   choices:
-    - None
-    - Docker
-    - Podman
+  - None
+  - Docker
+  - Podman
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:165]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 9]
```diff
@@ -178,9 +177,9 @@ code_of_conduct:
   default: None
   # first choice is the default for the UI
   choices:
-    - None
-    - contributor-covenant
-    - citizen-code-of-conduct
+  - None
+  - contributor-covenant
+  - citizen-code-of-conduct
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:182]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 10]
```diff
@@ -190,9 +189,9 @@ governance_document:
   default: None
   # first choice is the default for the UI
   choices:
-    - None
-    - numpy-governance
-    - sciml-governance
+  - None
+  - numpy-governance
+  - sciml-governance
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:194]: Ditto

File: src/scicookie/profiles/base.yaml [hunk 11]
```diff
@@ -202,8 +201,8 @@ roadmap_document:
   default: None
   # first choice is the default for the UI
   choices:
-    - None
-    - pytorch-ignite-roadmap
+  - None
+  - pytorch-ignite-roadmap
```
Comment (Saransh-cpp) [inline src/scicookie/profiles/base.yaml:205]: Ditto

File: tests/smoke/auto-format-tools.sh [hunk 12]
```diff
@@ -2,5 +2,4 @@
 
 SMOKE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
-. ${SMOKE_DIR}/base.sh "use_black=yes use_blue=no"
-. ${SMOKE_DIR}/base.sh "use_black=no use_blue=yes"
+. ${SMOKE_DIR}/base.sh "use_black=no"
```
Comment (Saransh-cpp) [inline tests/smoke/auto-format-tools.sh:5]: We should test both the cases
```suggestion
. ${SMOKE_DIR}/base.sh "use_black=no"
. ${SMOKE_DIR}/base.sh "use_black=yes"
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks for working on this, @Vishalk91-4! See my comments below -

File: CHANGELOG.md [hunk 13]
```diff
@@ -42,7 +42,7 @@
 ### Bug Fixes
 
 * Fix error in `__author__` introduced by [#156](https://github.com/osl-incubator/scicookie/issues/156) ([67a40af](https://github.com/osl-incubator/scicookie/commit/67a40afba6d0041b226e1c37ff46f252a36f843d))
-* Make `black` "yes" and `blue` "no" by default ([#164](https://github.com/osl-incubator/scicookie/issues/164)) ([88048e4](https://github.com/osl-incubator/scicookie/commit/88048e43165c5918f4f5e70e9a5b374f9d5d5cf2))
+* Make `black` "yes" default ([#164](https://github.com/osl-incubator/scicookie/issues/164)) ([88048e4](https://github.com/osl-incubator/scicookie/commit/88048e43165c5918f4f5e70e9a5b374f9d5d5cf2))
```
Comment (Saransh-cpp) [inline CHANGELOG.md:45]: You should add a new CHANGELOG entry and not edit an old one

File: README.md [hunk 14]
```diff
@@ -20,7 +20,7 @@ for a Python package.
   - Licenses supported: MIT, BSD 3 Clause, ISC License, Apache Software License 2.0, and GPL 3
   - Documentation engines: mkdocs, sphinx, jupyter-boook
   - Test library: pytest, hypothesis 
-  - Auto format code tool: blue and black
+  - Auto format code tool: ruff
```
Comment (Saransh-cpp) [inline README.md:23]: I don't think we support ruff as an auto formatter tool?

**Comment by Vishalk91-4** [comment]
@Saransh-cpp, I have applied the mentioned changes, could you check them now

File: src/scicookie/profiles/base.yaml (hunk 2 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:69]: I removed blue from here, so it is showing rest as added, I guess

File: src/scicookie/hooks/post_gen_project.py (hunk 1 above)
Comment (Vishalk91-4) [reply src/scicookie/hooks/post_gen_project.py:265]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 3 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:80]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 5 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:110]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 4 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:98]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 7 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:152]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 8 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:165]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 9 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:182]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 10 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:194]: Implemented

File: src/scicookie/profiles/base.yaml (hunk 11 above)
Comment (Vishalk91-4) [reply src/scicookie/profiles/base.yaml:205]: Implemented

File: tests/smoke/auto-format-tools.sh (hunk 12 above)
Comment (Vishalk91-4) [reply tests/smoke/auto-format-tools.sh:5]: Fixed it

File: CHANGELOG.md (hunk 13 above)
Comment (Vishalk91-4) [reply CHANGELOG.md:45]: I will add it, after this issue get's resolved

**Comment by Saransh-cpp** [comment]
I don't think you pushed your changes?

**Comment by Vishalk91-4** [comment]
> I don't think you pushed your changes?

I'm sorry, actually I was fixing it between my classes, so missed it

File: src/scicookie/{{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/__main__.py [hunk 15]
```diff
@@ -6,8 +6,7 @@
 - https://docs.python.org/2/using/cmdline.html#cmdoption-m
 - https://docs.python.org/3/using/cmdline.html#cmdoption-m
 """
-{%- if cookiecutter.use_blue == "yes" %}
-  {%- set QUOTE = "'" -%}
+
 {%- elif cookiecutter.use_black == "yes" %}
```
Comment (Saransh-cpp) [inline src/scicookie/{{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/__main__.py:10]: This should now be an `if` condition

File: src/scicookie/profiles/base.yaml (hunk 2 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:69]: Nope, the indentation is wrong. Maybe your IDE formatted them with the rules you defined.

File: src/scicookie/hooks/post_gen_project.py [hunk 16]
```diff
@@ -258,14 +257,6 @@ def http2ssh(url):
     return url.replace("/", ":", 1)
 
 
-def validation():
```
Comment (Saransh-cpp) [inline src/scicookie/hooks/post_gen_project.py:261]: Please check if this function is being called anywhere and remove all those occurences

File: src/scicookie/profiles/base.yaml (hunk 4 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:98]: Not resolved, marking this as unresolved, see the indentation

File: src/scicookie/profiles/base.yaml (hunk 5 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:110]: Ditto

File: src/scicookie/profiles/base.yaml (hunk 7 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:152]: Ditto

File: src/scicookie/profiles/base.yaml (hunk 8 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:165]: Ditto

File: src/scicookie/profiles/base.yaml (hunk 9 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:182]: Dittp

File: src/scicookie/profiles/base.yaml (hunk 10 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:194]: Ditto, please don't resolve conversations without solving them

File: src/scicookie/profiles/base.yaml (hunk 11 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:205]: Ditto

File: src/scicookie/profiles/base.yaml (hunk 3 above)
Comment (Saransh-cpp) [reply src/scicookie/profiles/base.yaml:80]: Ditto

File: src/scicookie/{{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/__main__.py (hunk 15 above)
Comment (Vishalk91-4) [reply src/scicookie/{{cookiecutter.project_slug}}/{{cookiecutter.package_slug}}/__main__.py]: fixed this

File: src/scicookie/{{cookiecutter.project_slug}}/.releaserc.json [hunk 17]
```diff
@@ -1,5 +1,3 @@
-{%- if cookiecutter.use_blue -%}
-  {%- set QUOTE = "'" -%}
 {%- elif cookiecutter.use_black -%}
```
Comment (Saransh-cpp) [inline src/scicookie/{{cookiecutter.project_slug}}/.releaserc.json:1]: For example, this should be `if` now

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @Vishalk91-4! You can look at the failing tests locally or by clicking the details button on GitHub to see what should improved in this PR before it is ready to merge.

Please let us know if you are stuck at a test and we can try debugging that locally!

**Comment by Vishalk91-4** [comment]
```
Makim file: .makim.yaml
/home/vishalk91-4/.local/lib/python3.10/site-packages/_pytest/config/__init__.py:331: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
UsageError: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --template
  inifile: /mnt/c/Users/Vishal - IIT BHU/Documents/GitHub/scicookie/pyproject.toml
  rootdir: /mnt/c/Users/Vishal - IIT BHU/Documents/GitHub/scicookie
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --template
  inifile: /mnt/c/Users/Vishal - IIT BHU/Documents/GitHub/scicookie/pyproject.toml
  rootdir: /mnt/c/Users/Vishal - IIT BHU/Documents/GitHub/scicookie

Traceback (most recent call last):
  File "/tmp/tmpe2nw53_k.makim", line 1, in <module>
    pytest --template -s -vv src/scicookie tests
 

  RAN: /home/vishalk91-4/.local/bin/xonsh /tmp/tmpe2nw53_k.makim

  STDOUT:


  STDERR:
  ```
  @Saransh-cpp, facing this error while running tests, tried deleting template, faced another error where it couldn't find cookiecutter

**Comment by Saransh-cpp** [comment]
These would be issues with your local setup. It would be tough for me to reproduce them locally. Are you able to use scicookie locally to create packages?

You can always see the error logs using GitHub Actions, for instance the failing `auto-format-tools.sh` test is here - https://github.com/osl-incubator/scicookie/actions/runs/7720857565/job/21046425608?pr=198

**Comment by Naman-Priyadarshi** [comment]
Hi @Vishalk91-4, are you currently working on this?

**Comment by Vishalk91-4** [comment]
> Hi @Vishalk91-4, are you currently working on this?

Not particularly, you can work on it now

**Pull request closed without merging**